internal/pkg/agent/cmd/run.go

```diff
@@ -579,6 +579,18 @@ func handleUpgrade() error {
 		return nil
 	}
 
+	if err := ensureInstallMarkerPresent(); err != nil {
+		return err
+	}
+
+	if err := ensureServiceConfigUpToDate(); err != nil {
```

Comment:
I think it would be good to have this as `upgrade.EnsureServiceConfig()`. Related to your ask below.

Reply:
Done in 09ec4a120a90bc3e01d92db93c5887a9fc56401f.